### PR TITLE
Don't use wmic's ProviderName if it's empty

### PR DESCRIPTION
**Pull request by zvin**
Change-type: patch

### CODE REVIEW

**Comment by lurch** [comment]
Is it worth adding another unit-test for this?

File: lib/gui/app/os/windows-network-drives.js [hunk 1]
```diff
@@ -107,6 +107,8 @@ const getWindowsNetworkDrives = async () => {
       // eslint-disable-next-line no-magic-numbers
       return [ str.slice(0, colonPosition + 1), _.trim(str.slice(colonPosition + 1)) ]
     })
+    // eslint-disable-next-line no-magic-numbers
+    .filter((couple) => couple[1].length > 0)
```
Comment (zrzka) [inline lib/gui/app/os/windows-network-drives.js]: I'd like to see a note somewhere why we're doing this. I mean, we know now, but in a couple of days ... This VMware Tools case seems like a special one worth documenting.

Comment (thundron) [reply lib/gui/app/os/windows-network-drives.js]: imho in a comment block over that line

**Comment by lurch** [comment]
In the `wmic-output.txt` it probably makes sense to use two different drive letters, rather than `Z:` both times? :wink:

File: lib/gui/app/os/windows-network-drives.js (hunk 1 above)
Comment (zvin) [reply lib/gui/app/os/windows-network-drives.js]: This is not vmwere specific.
The second part is supposed to be `\\<ip address>\folder`.
If it's an empty string we don't use it.

**Comment by zvin** [comment]
@lurch we use `Z:` and the test checks that the non empty value for `Z:`is being used.

**Comment by lurch** [comment]
But surely in any real wmic output, you'll never get the same drive-letter appearing twice? Feel free to ignore me if I'm just misunderstanding what the test is doing.

**Comment by lurch** [comment]
For future reference, this PR relates to #2858

**Pull request merged**